src/toil_scripts/lib/__init__.py

```diff
@@ -31,6 +36,17 @@ def partitions(l, partition_size):
         yield l[i:i + partition_size]
 
 
+def get_work_directory():
+    try:
+        return os.path.join(os.environ['TOIL_WORKDIR'], str(uuid4()))
+    except KeyError:
+        fallback_dir = '/mnt/ephemeral'
+        if os.path.exists(fallback_dir):
+            return os.path.join(fallback_dir, str(uuid4()))
+        else:
+            return tempfile.mkdtemp()
+
+
```

Comment:
But need to add docstring specifying the post-condition, i.e. directory exists when function returns.